test/functorch/test_eager_transforms.py

```diff
@@ -2388,6 +2388,35 @@ def f(x):
         with self.assertRaises(RuntimeError):
             grad(f)(x)
 
+    def test_autograd_function_debug_switch(self, device):
+        class MySin(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, x):
+                ctx.save_for_backward(x)
+                return x.sin()
+
+            @staticmethod
+            def backward(ctx, gy):
+                x, = ctx.saved_tensors
+                return gy * x.cos()
+
+        x = torch.randn([])
```

Comment:
Is this tensor intentionally doesn't require gradients? I guess it doesn't matter because should error anyway.

Reply:
The functorch grad transform doesn't use the requires_gradness of the tensor to compute gradients, it accepts an argnums argument that says which input to compute gradients with respect to

Reply:
Oh whoops, good point.